packages/shopify-cart/src/utils/transformShopifyLineToNacelleLine.ts

```diff
@@ -1,14 +1,33 @@
 import { CartLine, NacelleCartLine } from '../types/cart.type';
 
-export default function (shopifyLines: CartLine[]): NacelleCartLine[] {
-  return shopifyLines?.map((shopifyLine) => {
-    const transformedLine: CartLine = { ...shopifyLine };
-    const nacelleEntryId = transformedLine.nacelleEntryId?.value as string;
-    delete transformedLine.nacelleEntryId;
+export interface TransformShopifyCartLineParams {
+  lines: CartLine[];
+  shopifyShopId: string;
+  locale: string;
+}
+
+export default function ({
+  lines,
+  shopifyShopId,
+  locale
+}: TransformShopifyCartLineParams): NacelleCartLine[] {
+  return lines?.map((line) => {
+    const shopifyVariantId = line.merchandise.sourceEntryId
+      .split('gid://shopify/ProductVariant/')[1]
+      .replace('/', '');
+
+    let nacelleEntryId = `id://SHOPIFY/${shopifyShopId}/default/PRODUCT_VARIANT/${shopifyVariantId}/${locale}`;
+
+    if (typeof window !== 'undefined') {
+      nacelleEntryId = globalThis.btoa(nacelleEntryId);
+    } else {
+      nacelleEntryId = new Buffer(nacelleEntryId).toString('base64');
```

Comment:
```suggestion
      nacelleEntryId = Buffer.from(nacelleEntryId).toString('base64');
```

![buffer is deprecated](https://user-images.githubusercontent.com/5732000/192567679-44603d86-c9e0-4ca9-bcff-88dba3407d0b.png)

From https://nodejs.org/en/docs/guides/buffer-constructor-deprecation:

![new Buffer security risk](https://user-images.githubusercontent.com/5732000/192567363-40cee6a4-d114-456d-846b-2595b5521fc9.png)